integration-cli/docker_cli_build_test.go

```diff
@@ -5660,3 +5660,434 @@ func (s *DockerSuite) TestBuildNullStringInAddCopyVolume(c *check.C) {
 	_, err = buildImageFromContext(name, ctx, true)
 	c.Assert(err, check.IsNil)
 }
+
+func (s *DockerSuite) TestBuildBuildTimeArg(c *check.C) {
+	imgName := "bldargtest"
+	envKey := "foo"
+	envVal := "bar"
+	args := []string{
+		"--build-arg", fmt.Sprintf("%s=%s", envKey, envVal),
+	}
+	dockerfile := fmt.Sprintf(`FROM busybox
+		ARG %s=
+		RUN echo $%s
```

Comment:
Just for future reference, if you wrote:
```
RUN [ "$foo" = "bar" ]
```
then you could have the builder verify that "$foo" is set properly w/o needing to check the output for "bar". We do that in the builder testscases a lot.